packages/manager/src/features/Linodes/LinodeCreatev2/UserData/UserData.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useMemo } from 'react';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+
+import { Accordion } from 'src/components/Accordion';
+import { Link } from 'src/components/Link';
+import { Notice } from 'src/components/Notice/Notice';
+import { TextField } from 'src/components/TextField';
+import { Typography } from 'src/components/Typography';
+import { useRestrictedGlobalGrantCheck } from 'src/hooks/useRestrictedGlobalGrantCheck';
+import { useImageQuery } from 'src/queries/images';
+import { useRegionsQuery } from 'src/queries/regions/regions';
+
+import { atou, utoa } from '../../LinodesCreate/utilities';
+import { UserDataHeading } from './UserDataHeading';
+
+import type { CreateLinodeRequest } from '@linode/api-v4';
+
+export const UserData = () => {
+  const { control } = useFormContext<CreateLinodeRequest>();
+
+  const regionId = useWatch({ control, name: 'region' });
+  const imageId = useWatch({ control, name: 'image' });
+
+  const [formatWarning, setFormatWarning] = React.useState(false);
+
+  const { data: regions } = useRegionsQuery();
+  const { data: image } = useImageQuery(imageId ?? '', Boolean(imageId));
+
+  const checkFormat = ({
+    hasInputValueChanged,
+    userData,
+  }: {
+    hasInputValueChanged: boolean;
+    userData: string;
+  }) => {
+    const userDataLower = userData.toLowerCase();
+    const validPrefixes = ['#cloud-config', 'content-type: text/', '#!/bin/'];
+    const isUserDataValid = validPrefixes.some((prefix) =>
+      userDataLower.startsWith(prefix)
+    );
+    if (userData.length > 0 && !isUserDataValid && !hasInputValueChanged) {
+      setFormatWarning(true);
+    } else {
+      setFormatWarning(false);
+    }
```

Comment:
I think I was the one who wrote this code originally but looking at it again:

```suggestion
    setFormatWarning(userData.length > 0 && !isUserDataValid && !hasInputValueChanged);
```